### PR TITLE
Avalondock theme

### PR DESCRIPTION
**Pull request by flabbet**
- Redesigned AvalonDock with document previews.
- Added Layer previews
- Changed highlighted layer color
- (Hopefully) Improved performance of `CombineLayers`

### CODE REVIEW

**Comment by CPKreuz** [comment]
I recommend making sure that the tests are passing

**Comment by flabbet** [comment]
image

**Review by CPKreuz — CHANGES_REQUESTED** [review]
I now looked at it and I really like it

One thing I noticed is when you undock both Color Picker and Layers the docking thing is still there. Would be nice if it would collapse

rightNotGoingAway

**Comment by CPKreuz** [comment]
In other words, why aren't we using the same DockingManager for both document and color picker/layers

**Comment by CPKreuz** [comment]
Also, can you remove the corners between tabs and on the top left
And the close button is still blue

Screenshot (5)

**Comment by flabbet** [comment]
I fixed one CornerRadius below the document tabs, but the rest is, for now, a little too much work. I'll do that eventually. But for now, I would prefer to focus on more important things

File: PixiEditor/Models/ImageManipulation/BitmapUtils.cs [hunk 1]
```diff
@@ -73,6 +80,32 @@ public static WriteableBitmap CombineLayers(Layer[] layers, int width, int heigh
...
+            foreach (var layer in document.Layers.Where(x => x.IsVisible && x.Opacity > 0.8f))
+            {
+                previewBitmap.Blit(
+                    new Rect(layer.OffsetX, layer.OffsetY, layer.Width, layer.Height),
+                    layer.LayerBitmap,
+                    new Rect(0, 0, layer.Width, layer.Height));
+            }
+
+            int width = document.Width >= document.Height ? maxPreviewWidth : document.Width / (document.Height / maxPreviewHeight);
+            int height = document.Height > document.Width ? maxPreviewHeight : document.Height / (document.Width / maxPreviewWidth);
```
Comment (CPKreuz) [inline PixiEditor/Models/ImageManipulation/BitmapUtils.cs:104]: I get a `DivideByZeroException` for some images, maybe use floats for dividing

**Review by CPKreuz — CHANGES_REQUESTED** [review]
Middle Mouse button should also select the document

**Review by CPKreuz — APPROVED** [review]

**Pull request merged**